runelite-client/src/main/java/net/runelite/client/game/ClanManager.java

```diff
@@ -154,25 +150,18 @@ public void onClanChange(ClanChanged clanChanged)
 
 	private void loadClanChatIcons()
 	{
-		try
-		{
-			final IndexedSprite[] modIcons = client.getModIcons();
-			final IndexedSprite[] newModIcons = Arrays.copyOf(modIcons, modIcons.length + CLANCHAT_IMAGES.length);
-			int curPosition = newModIcons.length - CLANCHAT_IMAGES.length;
+		final IndexedSprite[] modIcons = client.getModIcons();
+		final IndexedSprite[] newModIcons = Arrays.copyOf(modIcons, modIcons.length + CLANCHAT_IMAGES.length);
+		int curPosition = newModIcons.length - CLANCHAT_IMAGES.length;
 
-			for (BufferedImage image : clanChatImages)
-			{
-				IndexedSprite sprite = createIndexedSprite(client, image);
-				newModIcons[curPosition++] = sprite;
-			}
-
-			client.setModIcons(newModIcons);
-			modIconsLength = newModIcons.length;
-		}
-		catch (IOException e)
+		for (BufferedImage image : clanChatImages)
```

Comment:
I'm not sure I can get rid of `clanChatImages` since it's used to implement `ClanManager.getClanImage`, but I definitely don't need to use two methods to implement this.

Reply:
ooh true, I always forget about that one. You can change that one to spriteManager.getSprite(CLANCHAT_IMAGES[.ordinal - 1], 0) i guess.

Reply:
Decided to keep `clanChatImages` around after all to avoid doing one resize/outline/toIndexedColor operation per call, which could be many per frame from the player indicators plugin. This loop will be over `CLANCHAT_IMAGES` instead as suggested.